lib/preload.js

```diff
@@ -276,8 +276,11 @@ class Preloader extends EventEmitter {
 		await new Bluebird((resolve, reject) => {
 			this.docker.modem.followProgress(
 				build,
-				(error) => {
+				(error, output) => {
 					// onFinished
+					if (!error && output) {
+						error = output.pop().error;
```

Comment:
Oh, I have already approved the PR and it is merged, but if it is possible for `output` to be an empty array, this could produce a TypeError:
```
$ node
> let output = []
> if (output) { output.pop().error }
Uncaught TypeError: Cannot read property 'error' of undefined
```
A fix, if needed, might be to test `if (!error && output && output.length)`.

Reply:
It's possible, I will open a new PR for the fixup!